web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,191 @@
+        <template v-for="commit in commits" :key="commit.id">
+          <div class="vertical item gt-df gt-fr gt-gap-2 gt-border-secondary-top" :class="{selected: commit.selected}" @click.exact="commitClicked(commit.id)" @click.shift.exact.stop.prevent="commitClickedShift(commit)">
+            <div class="gt-f1 gt-df gt-fc gt-gap-2">
+              <div class="gt-ellipsis commit-list-summary">
+                {{ commit.summary }}
+              </div>
+              <div class="gt-ellipsis text light-2">
+                {{ commit.committer_or_author_name }}
+                <span class="text right">
+                  <relative-time class="time-since" prefix="" :datetime="commit.time" data-tooltip-content data-tooltip-interactive="true">{{ commit.time }}</relative-time>
+                </span>
+              </div>
+            </div>
+            <div class="gt-mono">
+              {{ commit.short_sha }}
+            </div>
+          </div>
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {SvgIcon} from '../svg.js';
+
+export default {
+  components: {SvgIcon},
+  data: () => {
+    return {
+      menuVisible: false,
+      isLoading: false,
+      locale: {},
+      commits: [],
+      hoverActivated: false,
+      lastReviewCommitSha: null
+    };
+  },
+  computed: {
+    commitsSinceLastReview() {
+      if (this.lastReviewCommitSha) {
+        return this.commits.length - this.commits.findIndex((x) => x.id === this.lastReviewCommitSha) - 1;
+      }
+      return 0;
+    },
+    queryParams() {
+      return this.$el.parentNode.getAttribute('data-queryparams');
+    },
+    issueLink() {
+      return this.$el.parentNode.getAttribute('data-issuelink');
+    }
+  },
+  mounted() {
+    // Add a click listener to close this menu on click outside of this element
+    //  when the dropdown is currently visible opened
+    document.body.addEventListener('click', (event) => {
+      if (this.$el.contains(event.target)) return;
+      if (this.menuVisible) {
+        this.menuVisible = false;
+      }
+    });
+  },
+  methods: {
+    /** Opens our menu, loads commits before opening */
+    async toggleMenu() {
+      this.menuVisible = !this.menuVisible;
+      // load our commits when the menu is not yet visible (it'll be toggled after loading)
+      // and we got no commits
+      if (this.commits.length === 0 && this.menuVisible === true) {
+        this.isLoading = true;
+        await this.fetchCommits();
+        this.isLoading = false;
+      }
+    },
+    /** Load the commits to show in this dropdown */
+    async fetchCommits() {
+      const resp = await fetch(`${this.issueLink}/commits/list`);
+      const results = await resp.json();
+      this.commits.push(...results.commits);
+      this.commits.reverse();
+      this.lastReviewCommitSha = results.last_review_commit_sha !== '' ? results.last_review_commit_sha : null;
```

Comment:
```suggestion
      this.lastReviewCommitSha = results.last_review_commit_sha ? results.last_review_commit_sha : null;
```

Reply:
hm - when changing the code i get an eslint warning to prefer logical operator over ternary. https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v47.0.0/docs/rules/prefer-logical-operator-over-ternary.md

so i switched to a || instead.